diff --git a/lib/merge-config.js b/lib/merge-config.js
index 48bf3cf546e0c72803499a6ca20e27a3ebd4c4dd..b6478a261fe41aa15c11eaf8795f16c8eec5eb23 100644
--- a/lib/merge-config.js
+++ b/lib/merge-config.js
@@ -14,6 +14,12 @@ const readPkg = require('read-pkg')
 const readPkgUp = require('read-pkg-up')
 const { URL } = require('url')
 
+const hosts = {
+  github: require('../hosts/github'),
+  bitbucket: require('../hosts/bitbucket'),
+  gitlab: require('../hosts/gitlab')
+}
+
 const rhosts = /github|bitbucket|gitlab/i
 
 function semverTagsPromise (options) {
@@ -255,7 +261,7 @@ async function mergeConfig (options, context, gitRawCommitsOpts, parserOpts, wri
     }
 
     if (type) {
-      hostOpts = require('../hosts/' + type)
+      hostOpts = hosts[type];
 
       context = {
         issue: hostOpts.issue,
